daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -1482,6 +1503,78 @@ object Main {
         }
       }
 
+      case Some(conf.exi) => {
+        val exiOpts = conf.exi
+        val channel = exiOpts.output.toOption match {
+          case Some("-") | None => Channels.newChannel(STDOUT)
+          case Some(file) => new FileOutputStream(file).getChannel()
+        }
+        val output = Channels.newOutputStream(channel)
+
+        val inputStream = exiOpts.infile.toOption match {
+          case Some("-") | None => STDIN
+          case Some(file) => {
+            val f = new File(file)
+            new FileInputStream(f)
+          }
+        }
+        val input = new InputSource(inputStream)
+
+        val exiFactory = try {
+          if (exiOpts.schema.isDefined)
+            getExiFactoryOpt(InfosetType.EXISA, exiOpts.schema.toOption)
+          else
+            getExiFactoryOpt(InfosetType.EXI, exiOpts.schema.toOption)
+        } catch {
+          case e: EXIException => {
+            Logger.log.error("Error creating EXI grammar for the supplied schema: %s".format(Misc.getSomeMessage(e).get))
+            return ExitCode.Failure
```

Comment:
It might make things a bit more complicated, but we should avoid the use of `return`. Same with other uses of return below.